src/lib.rs

```diff
@@ -354,9 +355,15 @@ impl TxBuilder {
             drain_wallet: false,
             drain_to: None,
             rbf: None,
+            data: Vec::new(),
         }
     }
-
+    fn add_data(&self, data: Vec<u8>) -> Arc<Self> {
+        Arc::new(TxBuilder {
+            data,
+            ..self.clone()
+        })
+    }
```

Comment:
All other methods on the `TxBuilder` are in the order the struct is defined. That would put `add_data()` at the end, just before `finish()`. Unless we have an explicit other ordering system (I don't think we do in this case if I'm not mistaken), let's keep them in that order (note that it will also mirror the order in the UDL file).

Reply:
Acknowledged, I have made the requested change. For next time, I will make a note of not changing the sequence of the code less it requires explicit ordering system. Thank you for reviewing and testing. 🚀